### ISSUE TITLE
Add tensor padding options to improve hardware utilization

### ISSUE THREAD

**Issue by jon-tow** [CarperAI/trlx#284]
### 🚀 The feature, motivation, and pitch

Recent [discussion](https://twitter.com/karpathy/status/1621578354024677377) on Twitter highlighted the importance of tensor padding to improve hardware util. NeMo seems to already support GPU-friendly vocab size padding from discussions with @cat-state but we should also consider adding an optional [`pad_to_multiple_of`](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizer.__call__.pad_to_multiple_of) arg to our tokenizer calls with the accelerate/transformers backend to satisfy [Tensor Core requirements](https://docs.nvidia.com/deeplearning/performance/dl-performance-matrix-multiplication/index.html#requirements-tc).

![image](https://user-images.githubusercontent.com/41410219/217096562-32589fb5-817d-40fc-bd89-ccc909c40516.png)

The implementer should also provide system plots to display any improvements/findings.

**Comment by Rami-Ismael** [comment]
Can I do this? I have experience in building documents for single gpubinplemenation for the NLP model.

**Comment by jon-tow** [comment]
@Rami-Ismael Sure! Give it a go; I don't believe anyone else is working on this at the moment.

**Comment by Rami-Ismael** [comment]
I was wondering if it will best to add the *pad_to_multiple_of* as an argument in the Trainer Config?

**Comment by jon-tow** [comment]
We have a [`TokenizerConfig`](https://github.com/CarperAI/trlx/blob/206d885a2fbcbfd848b174714c96c1de903e4f54/trlx/data/configs.py#L63) that could be used to store this option for padding with the tokenizer (and if you wanted to add vocab size padding we could add that to `ModelConfig`). What do you think?

**Comment by Rami-Ismael** [comment]
I was looking at the code. My expectation was all the arguments of the tokenizer Config will be fed into the Hugging Face PreTrain Tokenizer parameter; this will produce an error as the Hugging Face Tokenizer call function has that as an argument.

The vocab size padding will be easy to implement as there is code implementation in [megatron project](https://github.com/NVIDIA/Megatron-LM/blob/285068c8108e0e8e6538f54fe27c3ee86c5217a2/megatron/tokenizer/tokenizer.py#L47)

**Comment by jon-tow** [comment]
The options in a `TokenizerConfig` get picked apart and used where needed. They're not all fed through the `AutoTokenizer.from_pretrained` as a glob of kwargs, e.g. see how the tokenizer objects are instantiated and assigned fields: 
https://github.com/CarperAI/trlx/blob/206d885a2fbcbfd848b174714c96c1de903e4f54/trlx/trainer/accelerate_base_trainer.py#L62-L66
So if the `pad_to_multiple_of` is in `TokenizerConfig` we might be able to pass it around to `tokenizer()` calls?

**Comment by Rami-Ismael** [comment]
For some RL Algorithm that uses the tokenized dialogue 
https://github.com/CarperAI/trlx/blob/206d885a2fbcbfd848b174714c96c1de903e4f54/trlx/pipeline/offline_pipeline.py#LL25-L28C38
which is only ILQL.
In the **accelerate_base_trainer.py**, there is no *tokenizer ()* calls.

**Comment by Rami-Ismael** [comment]
Can you add this into this context https://twitter.com/cHHillee/status/1630274804795445248. This issue is being referenced. The Twitter post is from an expert PyTorch, and hardware utilization gives more context to the Andew Karpathy tweet.
